plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseSharedMetastoreTest.java

```diff
@@ -128,4 +134,53 @@ public void testShowSchemas()
                 showCreateIcebergWithRedirectionsSchema,
                 getExpectedIcebergCreateSchema("iceberg_with_redirections"));
     }
+
+    @Test
+    public void testTimeTravelWithRedirection()
+            throws InterruptedException
+    {
+        assertQuery("SELECT * FROM iceberg_with_redirections." + schema + ".nation", "SELECT * FROM nation");
+        long snapshot1 = getLatestSnapshotId();
+        long v1EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot1);
+        Thread.sleep(1);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(25, 'POLAND', 3, 'test 1')", schema), 1);
+        long snapshot2 = getLatestSnapshotId();
+        long v2EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot2);
+        Thread.sleep(1);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(26, 'CHILE', 1, 'test 2')", schema), 1);
+        long snapshot3 = getLatestSnapshotId();
+        long v3EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot3);
+        long incorrectSnapshot = 2324324333L;
+        Thread.sleep(1);
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, snapshot1), "SELECT * FROM nation");
```

Comment:
I should have been clearer in the issue, but I think the more interesting thing to test here is that queries on iceberg tables going through `hive_with_redirections` work.

Also that something like `select * from iceberg_with_redirection.schema.a_hive_tabl for timestamp as of ...` should fail